src/Entities/Question.php

```diff
@@ -47,6 +49,16 @@ public function set_data(BaseQuestionType $data)
         $this->data = $data;
     }
 
+    public function set_item_reference($itemReference)
+    {
+        $this->itemReference = $itemReference;
+    }
+
+    public function get_item_reference()
+    {
+        return $this->itemReference;
+    }
+
     /**
      * @override
      **/
```

Comment:
@ttton  I am not getting the point for changing the to_array() method.
We are getting the following output if we call the to_array -
`Array
(
    [reference] => 594fa96e-135b-4788-86e9-2f59f03c23bb
    [type] => mcq
    [data] => Array
        (
            [stimulus] => <p>[This is the stem.]</p>
            [type] => mcq
            [ui_style] => Array
                (
                    [type] => horizontal
                )

            [validation] => Array
                (
                    [scoring_type] => exactMatch
                    [valid_response] => Array
                        (
                            [score] => 1
                        )

                )

            [options] => Array
                (
                    [0] => Array
                        (
                            [value] => 0
                            [label] => [Choice A]
                        )

                    [1] => Array
                        (
                            [value] => 1
                            [label] => [Choice B]
                        )

                    [2] => Array
                        (
                            [value] => 2
                            [label] => [Choice C]
                        )

                    [3] => Array
                        (
                            [value] => 3
                            [label] => [Choice D]
                        )

                )

        )

    [widgetType] => response
    [itemReference] => PAPE_31
)
`

Reply:
Hi @rakeshqdsa, that's what I mean - it's including the `itemReference` which shouldn't be included in the serialized data of a Question entity, and the `widgetType` key needs to be `widget_type`.

Reply:
I have changed the widgetType to widget_type .
But itemreference is used to set assessment item's label in xml file . That's why we modified this entity and include this itemreference in entity.

Reply:
@rakeshqdsa Right, ok that's fair enough. I didn't realize that the item reference was being serialized anywhere (thought it was directly being accessed).

In that case, I'm happy with this - I would prefer consistency of style within a class though (if possible), so what do you think about actually converting the `itemReference` instance variable to `item_reference` for this class? It's unconventional, but since we serialize it this way it sort of makes more sense.

If this impacts other code outside this PR then I'd say don't worry about it.

Reply:
Done